CS_UCDs.ipynb

```diff
@@ -0,0 +1,271 @@
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "tables = hsc.service.tables  # Queries for details of the service's tables\n",
+    "print(f'{len(tables)} tables:')\n",
+    "for t in tables:\n",
+    "    print(f'{t.name:30s} - {t.description}\\n----')  # A more succinct option than t.describe()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let's look at the first 10 columns of the DetailedCatalog table.  Again, note that calling the columns attribute sends another query to the service to ask for the columns."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "columns=tables['dbo.DetailedCatalog'].columns\n",
+    "for c in columns:\n",
+    "    #print(f'{c.name:30s} - {c.description}')\n",
+    "    print(\"{} - {}\".format( \"{} [{}]\".format(c.name,c.ucd).ljust(30) , c.description))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The PyVO method to get the columns will automatically fetch all the meta-data about those columns.  It's up to the service provider to set them correctly, of course, but in this case, we see that the column named \"MatchRA\" is identified with the UCD \"pos.eq.ra\".  \n",
+    "\n",
+    "So if we did not know the exact name used in HSCv3 for the RA, we could do something like this looking for the string \"RA\":"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "ra_name=[c.name for c in columns if 'RA' in c.name or \"ascension\" in c.name.lower()]\n",
+    "print(ra_name)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "But a more general approach is to check for the correct UCD.  It also has the further advantage that it can be used to label columns that should be used for certain purposes when there are multiple possibilities.  For instance, this table has MatchRA and SourceRA.  Let's check the UCD:  \n",
+    "\n",
+    "(Note that the UCD is not required.  If it isn't there, you get a None type, so code the check carefully)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "ra_name=[c.name for c in columns if c.ucd and 'pos.eq.ra' in c.ucd][0]\n",
+    "dec_name=[c.name for c in columns if c.ucd and 'pos.eq.dec' in c.ucd][0]\n",
+    "ra_name,dec_name"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "What that shows you is that though there are two columns in this table that give RA information, only one has the 'pos.eq.ra' UCD. The documentation for this ought to explain the usage of these columns, and the UCD should not be used as a substitute for understanding the table. But it can be a useful tool.  "
```

Comment:
There is some additional complexity that we'll eventually have to deal with here (possibly with new astropy or pyvo features?).

There can be more than one RA and Dec in a catalog record.  The "main" ones are supposed to be indicated by adding another component to the UCD, so these are the recommended UCDs for the main equatorial position in a record: `pos.eq.ra;meta.main` and `pos.eq.dec;meta.main`

Sadly the HSC catalog used in the example is not using these, and it should be corrected.

Another problem is that many query results still use the older style of UCDs.  For the main equatorial positions in a record, these are `POS_EQ_RA_MAIN` and `POS_EQ_DEC_MAIN`.  It would be too much to deal with all old-style UCDs, but for position (and possibly `ID_MAIN` vs. `meta.id;meta.main`) it's probably worth it.  The CFAZ cone search results from the Catalogs notebook use these:
```
import pyvo as vo
import astropy.units as u
from astropy.coordinates import SkyCoord

services = vo.regsearch(servicetype='scs', keywords=['zcat Whipple Observatory UGC'])
coord = SkyCoord.from_name("m51")
results = services[0].search(pos=coord, radius=10*u.arcmin)
for fd in results.fielddescs:
    print(fd)
```

Reply:
> Another problem is that many query results still use the older style of UCDs. For the main equatorial positions in a record, these are `POS_EQ_RA_MAIN` and `POS_EQ_DEC_MAIN`.

I realized this isn't totally random.  The Simple Cone Search and Simple Image Access (v1.0) protocols actually require the use of `POS_EQ_RA_MAIN` and `POS_EQ_DEC_MAIN`.  The standards are here:

SCS - http://www.ivoa.net/documents/REC/DAL/ConeSearch-20080222.html#req
SIA 1.0 - http://www.ivoa.net/documents/SIA/20091116/REC-SIA-1.0.html#query-out

Reply:
So the SCS document is inconsistent with the UCD definition doc?  That's not going to be pleasant to handle and explain.

Reply:
Yes, I'm wondering if we can handle this with some kind of wrapper in in pyvo or astropy, but it's not trivial.

Reply:
I've just added this to the KNOWN_ISSUES.